src/main/java/com/fouristhenumber/utilitiesinexcess/common/blocks/BlockCursedEarth.java

```diff
@@ -72,19 +86,28 @@ public ArrayList<ItemStack> getDrops(World world, int x, int y, int z, int metad
     public void updateTick(World world, int x, int y, int z, Random random) {
         super.updateTick(world, x, y, z, random);
         if (world.isRemote) return;
+        trySpawnMob(world, x, y, z, random);
+        tryBurn(world, x, y, z, random);
+    }
+
+    public void trySpawnMob(World world, int x, int y, int z, Random random) {
         if (!world.isAirBlock(x, y + 1, z)) return;
-        if (world.getBlockLightValue(x, y + 1, z) >= 8) return;
+        int light = world.getBlockLightValue(x, y + 1, z);
+        if ((light >= 8 && !blessed) || light < 8 && blessed) return;
         if (!world.getGameRules()
             .getGameRuleBooleanValue("doMobSpawning")) return;
-        if (world.difficultySetting == EnumDifficulty.PEACEFUL) return;
-        if (random.nextInt(100) >= CursedEarthConfig.cursedEarthSpawnRate) return;
+        if (world.difficultySetting == EnumDifficulty.PEACEFUL && !blessed) return;
+        if (random.nextInt(100) + 1
+            >= (100 - (blessed ? CursedEarthConfig.blessedEarthSpawnRate : CursedEarthConfig.cursedEarthSpawnRate)))
```

Comment:
```suggestion
        if (random.nextInt(100)
            >= (100 - (blessed ? CursedEarthConfig.blessedEarthSpawnRate : CursedEarthConfig.cursedEarthSpawnRate)))
```

If the chance is 0%, this becomes `random.nextInt(100) + 1 >= 100`, which is still possible if `random.nextInt(100)` rolls a 99.